pylate/hf_hub/model_card.py

```diff
@@ -0,0 +1,259 @@
+from __future__ import annotations
+
+import logging
+from collections import defaultdict
+from dataclasses import dataclass, field, fields
+from pathlib import Path
+from platform import python_version
+from typing import TYPE_CHECKING, Any, Literal
+
+import torch
+import transformers
+from huggingface_hub import ModelCard
+from sentence_transformers import SentenceTransformerModelCardData
+from sentence_transformers import __version__ as sentence_transformers_version
+from sentence_transformers.util import (
+    is_accelerate_available,
+    is_datasets_available,
+)
+from torch import nn
+from transformers.integrations import CodeCarbonCallback
+
+from ..__version__ import __version__ as pylate_version
+
+if is_datasets_available():
+    pass
+
+logger = logging.getLogger(__name__)
+
+if TYPE_CHECKING:
+    from sentence_transformers.evaluation.SentenceEvaluator import SentenceEvaluator
+    from sentence_transformers.SentenceTransformer import SentenceTransformer
+    from sentence_transformers.trainer import SentenceTransformerTrainer
+
+
+IGNORED_FIELDS = ["model", "trainer", "eval_results_dict"]
+
+
+def get_versions() -> dict[str, Any]:
+    versions = {
+        "python": python_version(),
+        "sentence_transformers": sentence_transformers_version,
+        "transformers": transformers.__version__,
+        "torch": torch.__version__,
+        "pylate": pylate_version,
+    }
+    if is_accelerate_available():
+        from accelerate import __version__ as accelerate_version
+
+        versions["accelerate"] = accelerate_version
+    if is_datasets_available():
+        from datasets import __version__ as datasets_version
+
+        versions["datasets"] = datasets_version
+    from tokenizers import __version__ as tokenizers_version
+
+    versions["tokenizers"] = tokenizers_version
+    return versions
+
+
+@dataclass
+class PylateModelCardData(SentenceTransformerModelCardData):
+    """A dataclass storing data used in the model card.
+
+    Args:
```

Comment:
Could you update the docstring to match our docstring template (it will be used in the doc)